### PR TITLE
feat: Quick Stock  Balance

### PR DESCRIPTION
**Pull request by marination**
- It will display the stock balance and stock value given the date and warehouse
- Item code can be fed or barcode can be scanned to retrieve the values

### CODE REVIEW

File: erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2019, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Stock Quick Balance', {
+
+	setup: (frm) => {
+		frm.set_query('item', () => {
+			if (!frm.doc.warehouse) {
+				frm.trigger('warehouse_check');
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.js]: check_warehouse

File: erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+			}
+		});
+	},
+
+	make_custom_button: (frm) => {
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.js]: add context

File: erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+class StockQuickBalance(Document):
+	pass
+
+@frappe.whitelist()
+def get_item_stock_details(item, warehouse, date):
+	out = {}
+	barcodes = frappe.db.get_values("Item Barcode", filters={
+		"parent": item}, fieldname=["barcode"])
+
+	out["item_barcode"] = [x[0] for x in barcodes]
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.py]: don't fetch

File: erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	out["qty"] = get_stock_balance(item, warehouse, date)
+	out["value"] = get_stock_value_on(warehouse, date, item)
+	out["image"] = frappe.db.get_value(
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.py]: fetch_from

File: erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+from frappe import _
+from frappe.utils import cstr
+from frappe.model.document import Document
+from erpnext.stock.utils import get_stock_balance, get_stock_value_on
+
+class StockQuickBalance(Document):
+	pass
+
+@frappe.whitelist()
+def get_item_stock_details(item, warehouse, date):
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.py]: get stock_balance

File: erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	},
+
+	item_barcode: (frm) => {
+		if (!frm.doc.warehouse) {
+			frm.trigger('warehouse_check');
+		}
+
+		else if (frm.doc.item_barcode) {
+			frappe.call({
+				method: 'erpnext.stock.doctype.stock_quick_balance.stock_quick_balance.get_barcode_stock_details',
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.js]: get_barcode_stock_details should also return stock balance as well

File: erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.json [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+  {
+   "fieldname": "item_barcode",
+   "fieldtype": "Data",
+   "label": "Item Barcode"
+  },
+  {
+   "fetch_from": "item.item_name",
+   "fieldname": "item_name",
+   "fieldtype": "Data",
+   "label": "Item Name"
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.json]: read only

File: erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.json [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+ ],
+ "hide_toolbar": 1,
+ "in_create": 1,
+ "issingle": 1,
+ "modified": "2019-09-18 13:34:03.619283",
+ "modified_by": "Administrator",
+ "module": "Stock",
+ "name": "Stock Quick Balance",
+ "owner": "Administrator",
+ "permissions": [
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_quick_balance/stock_quick_balance.json]: add permissions for stock user and stock manager roles

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Review by rmehta — CHANGES_REQUESTED** [review]
Bad naming! There must be a way to define such names.
 `Get Quick Stock Balance` will be a better name.

**Review by nabinhait — APPROVED** [review]

**Pull request merged**